src/components/EditGlossary.jsx

```diff
@@ -7,38 +8,91 @@ import ErrorStyled from "./shared/Error";
 import ContainerStyled from "./shared/Container";
 import TabContainer from "./shared/TabContainer";
 
+import chromeStore from "../utils/chromeStore";
+
+import {
+  createBucket,
+  createGlossaryFromGoogleTranslation,
+  getCsvFormGooleStorage,
+  getGlossaryFormServer,
+  updateCsvFromGoogleStorage,
+  updateGlossaryFromServer,
+} from "../services/glossaryService";
+
 const initWordsToAdd = {
   text: "",
   translation: "",
 };
 
 function EditGlossary() {
   const [isLoading, setIsLoading] = useState(true);
-  const [error, setError] = useState(null);
+  const [errorMassage, setErrorMassage] = useState(null);
   const [user, setUser] = useState(null);
-  const [glossary, setGlossary] = useState(null);
+  const [glossary, setGlossary] = useState({});
+  const [isCreateBucket, setIsCreateBucket] = useState(null);
+  const [bucketId, setBucketId] = useState(null);
   const [wordsToAdd, setWordsToAdd] = useState(initWordsToAdd);
 
   const focus = useRef(null);
+  const history = useHistory();
 
   useEffect(() => {
-    chrome.storage.sync.get(["userData"], async ({ userData }) => {
-      if (chrome.runtime.lastError) {
+    (async () => {
+      try {
+        const userData = await chromeStore.get("userData");
+
+        setBucketId(userData.email.replace(/@|\./gi, ""));
+        setUser(userData);
+      } catch (error) {
         setIsLoading(false);
-        setError(chrome.runtime.lastError.message);
-        return;
+        setErrorMassage(error);
       }
-
-      setUser(userData);
-      setGlossary(userData.glossary);
-      setIsLoading(false);
-    });
+    })();
   }, []);
 
+  useEffect(() => {
+    if (user) {
+      const {
+        isServerOn,
+        email,
+        tokens: { accessToken },
+      } = user;
+
+      (async () => {
+        const { hasBucket, glossayData } = await getCsvFormGooleStorage(
+          {
+            accessToken,
+            bucketId,
+          },
+          setErrorMassage,
+        );
+
+        setIsCreateBucket(hasBucket);
+
+        if (isServerOn && isCreateBucket) {
```

Comment:
isCreateBucket의 상태 이름이 어색하고 이해하기 어려워서 한 쪽으로 이름을 통일해야 할 것 같습니다.
제 생각에는 예를 들어, 컴포넌트 상태 이름으로 hasBucket, API 요청으로 받는 결과 이름이 bucketExistence 이면 괜찮지 않을까?? 생각해봅니다.
나중에 철희님의 결정을 들려주세요!

예시
```suggestion
        const { hasBucket: bucketExistence, glossayData } = await getCsvFormGooleStorage(
          {
            accessToken,
            bucketId,
          },
          setErrorMassage,
        );

        setHasBuckett(bucketExistence);

        if (isServerOn && !hasBucket) {
```

Reply:
```
const dataFromGoogle = await getCsvFromGoogleStorage(
          {
            accessToken,
            bucketId,
          },
          setErrorMassage,
        );

        setHasBucket(dataFromGoogle.hasBucket);
```
동의어가 있는 것 또한 혼란의 여지가 있을 것 같아서 hasBucket으로 통일했습니다.